ballista/rust/scheduler/src/lib.rs

```diff
@@ -606,120 +589,65 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> SchedulerGrpc
-                    tonic::Status::internal(msg)
-                })?;
-                let num_tasks = task_status.len();
-                for task_status in task_status {
-                    self.state
-                        .save_task_status(&task_status)
-                        .await
-                        .map_err(|e| {
-                            let msg = format!("Could not save task status: {}", e);
-                            error!("{}", msg);
-                            tonic::Status::internal(msg)
-                        })?;
-                    if task_status.partition_id.is_some() {
-                        jobs.insert(task_status.partition_id.unwrap().job_id.clone());
-                    }
-                }
-                let mut executor_data = self
-                    .state
-                    .get_executor_data(&metadata.id)
-                    .await
-                    .map_err(|e| {
-                        let msg = format!(
-                            "Could not get metadata data for id {:?}: {}",
-                            &metadata.id, e
-                        );
-                        error!("{}", msg);
-                        tonic::Status::internal(msg)
-                    })?;
-                executor_data.available_task_slots += num_tasks as u32;
+            let num_tasks = task_status.len();
+            for task_status in task_status {
                 self.state
-                    .save_executor_data(executor_data)
+                    .save_task_status(&task_status)
                     .await
                     .map_err(|e| {
-                        let msg = format!("Could not save metadata data: {}", e);
+                        let msg = format!("Could not save task status: {}", e);
                         error!("{}", msg);
                         tonic::Status::internal(msg)
                     })?;
-                lock.unlock().await;
-            }
-            let tx_job = self.scheduler_env.as_ref().unwrap().tx_job.clone();
-            for job_id in jobs {
-                tx_job.send(job_id).await.unwrap();
+                if task_status.task_id.is_some() {
+                    jobs.insert(task_status.task_id.unwrap().job_id.clone());
+                }
             }
-            Ok(Response::new(UpdateTaskStatusResult { success: true }))
-        } else {
-            warn!("Received invalid task status update request");
-            Err(tonic::Status::invalid_argument(
-                "Missing metadata or task status in request",
-            ))
+            let mut executor_data = self.state.get_executor_data(&executor_id).unwrap();
+            executor_data.available_task_slots += num_tasks as u32;
+            self.state.save_executor_data(executor_data);
+        }
+        let tx_job = self.scheduler_env.as_ref().unwrap().tx_job.clone();
+        for job_id in jobs {
+            tx_job.send(job_id).await.unwrap();
```

Comment:
```suggestion
            tx_job.send(job_id).await?;
```

Reply:
Should we handle panic here?

Reply:
Thanks @Ted-Jiang. Agree. It's better to use ? instead of directly use unwrap for panic.